src/handlers.py

```diff
@@ -28,6 +22,42 @@ def on_delete(spec,uid,body,name,logger=None, **_):
     except KeyError as k:
         logger.info(f" This csec were not found in memory, maybe it was created in another run: {k}")
 
+@kopf.on.field('clustersecret.io', 'v1', 'clustersecrets', field='matchNamespace')
+def on_field_match_namespace(old, new, name, namespace, body, uid, logger=None, **_):
+    logger.debug(f'Namespaces changed: {old} -> {new}')
+
+    if old is not None:
+        logger.debug(f'Updating Object body == {body}')
+
+        try:
+            syncedns = body['status']['create_fn']['syncedns']
+        except KeyError:
+            logger.error('No Synced or status Namespaces found')
+            syncedns = []
+
+        v1 = client.CoreV1Api()
+        updated_matched = get_ns_list(logger, body, v1)
+        to_add = set(updated_matched).difference(set(syncedns))
+        to_remove = set(syncedns).difference(set(updated_matched))
+
```

Comment:
```suggestion
logger.debug(f'To add: {to_add}, To remove: {to_remove}')

```

Reply:
Great suggestion! The automatically committed suggestion had a weird indentation level so I jumped at the chance to write the message a bit more verbosely with:
```
logger.debug(f'Add secret to namespaces: {to_add}, remove from: {to_remove}')
```